crates/env/src/api.rs

```diff
@@ -526,6 +526,28 @@ where
     })
 }
 
+/// Returns the buffer back to the caller of the executed contract.
+///
+/// # Note
+///
+/// This function  stops the execution of the contract immediately.
+pub fn return_value_scoped(return_flags: ReturnFlags, return_value: &[u8]) -> ! {
+    <EnvInstance as OnInstance>::on_instance(|instance| {
+        EnvBackend::return_value_scoped(instance, return_flags, return_value)
+    })
+}
+
+/// Returns the buffer back to the caller of the executed contract.
+///
+/// # Note
+///
+/// This function  stops the execution of the contract immediately.
+pub fn input_scoped(buffer: &mut [u8]) {
```

Comment:
Very confusing name for the reason already described in another comment. I would rename this to `input_bytes` since that is what is actually is.